### PR TITLE
feat(swap): add screens to swap via Alex

### PR DESCRIPTION
**Pull request by zhigang1992**
This RP will enable swap feature within the xverse-web-extension directly.

- [x] Add swap screen
- [x] Add alex-sdk integration
- [x] Broadcast the actual swap transaction.
- [x] Integrate with stacks-transaction-sponsor "https://sponsor2.xverse.app"

# PR Type
- [ ] Bugfix
- [x] Enhancement
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

# Screenshot / Video

https://github.com/secretkeylabs/xverse-web-extension/assets/6109710/7d930f3b-86ae-4b32-9cff-b79c53eedb92

image

### CODE REVIEW

**Comment by zhigang1992** [comment]
Hey, can we update alex-sdk to 0.0.14?

We fixed an issue where some token pair might get routed to the older pools.

**Comment by teebszet** [comment]
> Hey, can we update alex-sdk to 0.0.14?
> 
> We fixed an issue where some token pair might get routed to the older pools.

do you mean 0.1.14?

**Review by teebszet — APPROVED** [review]

File: src/app/screens/swap/useSwap.test.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
+import { Currency } from 'alex-sdk';
+
+// TODO tim: fix jest transpiler to be able to import xverse-core
```
Comment (Imamah-Zafar) [inline src/app/screens/swap/useSwap.test.ts:3]: for the initial release , do we want to go with the copied function or is this in progress?

Comment (teebszet) [reply src/app/screens/swap/useSwap.test.ts:3]: let me make a follow up task to fix this test file, but essentially, I was testing a duplicated function to help with TDD.

Comment (teebszet) [reply src/app/screens/swap/useSwap.test.ts:3]: https://linear.app/xverseapp/issue/ENG-2474/fix-useswap-test-file

**Review by Imamah-Zafar — DISMISSED** [review]

**Comment by teebszet** [comment]
@Imamah-Zafar please reapprove after I updated the xverse-core version and xverse sponsor 2 host 🙏

**Review by Imamah-Zafar — DISMISSED** [review]

**Comment by teebszet** [comment]
@Imamah-Zafar sorry, added one more commit to fix after my merge commit 🙏

**Review by Imamah-Zafar — APPROVED** [review]

**Pull request merged**